opengrok-web/src/main/java/org/opengrok/web/PageConfig.java

```diff
@@ -494,6 +477,26 @@ public String canProcess() {
         return dirFileList;
     }
 
+    private Comparator<File> getFileComparator() {
+        if (getEnv().getListDirsFirst()) {
+            return (f1, f2) -> {
+                if (f1.isDirectory() && !f2.isDirectory()) {
+                    return -1;
+                } else if (!f1.isDirectory() && f2.isDirectory()) {
+                    return 1;
+                } else {
+                    return f1.getName().compareTo(f2.getName());
+                }
+            };
+        } else {
+            return Comparator.comparing(File::getName);
+        }
+    }
+
+    List<String> getSortedFiles(File[] files) {
```

Comment:
I think this is the good candidate for the `@VisibleForTesting` annotation we talked about :) 

https://javadoc.io/doc/org.jetbrains/annotations/latest/org/jetbrains/annotations/VisibleForTesting.html